source/lac/sparse_direct.cc

```diff
@@ -1359,14 +1370,47 @@ SparseDirectMUMPS::vmult(VectorType &dst, const VectorType &src) const
       // Copy solution into the given vector
       // For MUMPS with centralized solution (icntl[20]=0), the solution is only
       // on the root process (0) and needs to be distributed to all processes
-      if (Utilities::MPI::this_mpi_process(mpi_communicator) == 0)
+
+      // Get locally owned range for this process
+      const IndexSet &locally_owned = dst.locally_owned_elements();
+      const size_type local_size    = locally_owned.n_elements();
+
+      std::vector<double> local_values(local_size);
+
+      const unsigned int my_rank =
+        Utilities::MPI::this_mpi_process(mpi_communicator);
+
+      int my_size_int  = local_size;
+      int my_start_int = locally_owned.nth_index_in_set(0);
```

Comment:
These types aren't right. You don't actually need the `my_size_int` variable at all, since you already gave that variable a name (`local_size`). For `my_start_int`, I think it would be better to just put the call to `locally_owned.nth_index_in_set(0);` into the `gather()` call so that the compiler figures out the correct data type.